### PR TITLE
clang-tidy/implicit-conversion

### PR DESCRIPTION
**Pull request by rex-schilasky**
**Pull request type**

Please check the type of change your PR introduces:
- [X] Other (please describe): 
Clang-Tidy

**What is the current behavior?**
Clang-tidy warnings.

**What is the new behavior?**
Fixed clang-tidy warning: implicit conversion 'XYZ' (aka 'type_xyz') -> bool

**Does this introduce a breaking change?**

- [X] No

### CODE REVIEW

File: ecal/core/src/ecal_event.cpp [hunk 1]
```diff
@@ -42,7 +42,7 @@ namespace eCAL
     if(event_ == nullptr) return(false);
     EventHandleT event;
     event.name   = event_name_;
-    event.handle = ::CreateEvent(nullptr, false, false, event_name_.c_str());
+    event.handle = ::CreateEvent(nullptr, FALSE, FALSE, event_name_.c_str());
```
Comment (KerstinKeller) [inline ecal/core/src/ecal_event.cpp:45]: Why is `false` in capital letters? is that some kind of define for the C api or somethin?

File: ecal/core/src/io/win32/ecal_memfile_os.cpp [hunk 2]
```diff
@@ -89,13 +89,13 @@ namespace eCAL
 
       bool UnMapFile(SMemFileInfo& mem_file_info_)
       {
-        if (mem_file_info_.mem_address)
+        if (mem_file_info_.mem_address != nullptr)
         {
           UnmapViewOfFile(mem_file_info_.mem_address);
```
[the remark this answers is not in the document]
Comment (KerstinKeller) [reply ecal/core/src/io/win32/ecal_memfile_os.cpp:94]: https://github.com/eclipse-ecal/ecal/blob/1d2eeba3534a18d862fc985d65e1995ad346b9ac/ecal/core/src/io/ecal_memfile_info.h#L35
and 
https://github.com/eclipse-ecal/ecal/blob/1d2eeba3534a18d862fc985d65e1995ad346b9ac/ecal/core/src/io/ecal_memfile_info.h#L42

File: ecal/core/src/ecal_event.cpp (hunk 1 above)
Comment (rex-schilasky) [reply ecal/core/src/ecal_event.cpp:45]: FALSE is defined as 0 and the API is also expecting a 0, so clang states a bool -> int conversion

Comment (FlorianReimold) [reply ecal/core/src/ecal_event.cpp:45]: Somewhere in the win32 headers, there is
`#define FALSE 0`
`#define TRUE 1`

This is kind of necessary for the readability of old pure-C APIs like Win32. passing `true` or `false` will case an implicit bool->int conversion, which was desired here, but triggers the clang-tidy warning.

**Review by FlorianReimold — CHANGES_REQUESTED** [review]
We need to go over the new clang-tidy warnings again and solve the `int != nullptr` warnings

File: ecal/core/src/io/win32/ecal_memfile_os.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (rex-schilasky) [reply ecal/core/src/io/win32/ecal_memfile_os.cpp:94]: But the mem_address is a `void*` on all platforms

```cpp
#ifdef ECAL_OS_LINUX

typedef int  MemFileT;
typedef int  MapRegionT;

#endif /* ECAL_OS_LINUX */

namespace eCAL
{
  struct SMemFileInfo
  {
    SMemFileInfo()
    {
      refcnt      = 0;
      remove      = false;
      memfile     = 0;
      map_region  = 0;
      mem_address = 0;
      size        = 0;
      exists      = false;
    }
    int          refcnt;
    bool         remove;
    MemFileT     memfile;
    MapRegionT   map_region;
    void*        mem_address;
    std::string  name;
    size_t       size;
    bool         exists;
  };
}

```

[the remark this answers is not in the document]
Comment (KerstinKeller) [reply ecal/core/src/io/win32/ecal_memfile_os.cpp:94]: sorry, this applices to the `map_region` in the next clang-tidy comment.

**Pull request closed without merging**